src/grid/cell.component.ts

```diff
@@ -162,11 +160,12 @@ export class IgxGridCellComponent implements IGridBus, OnInit {
     @HostBinding("class.igx-grid__th--pinned-last")
     get isLastPinned() {
         const pinnedCols = this.grid.pinnedColumns;
-        if (pinnedCols.length === 0) {
-            return false;
-        } else {
-            return pinnedCols.indexOf(this.column) === pinnedCols.length - 1;
-        }
+        return pinnedCols[pinnedCols.length - 1] === this.column;
```

Comment:
Simplified this line because it is safe to access whatever element of an array, you just get undefined if the index is unavailable (as in the case when the array is empty)